### PR TITLE
Issue #10: Adding validation logic to guesses

### PR DESCRIPTION
**Pull request by tianshi-li**
Adding validation logic to guesses

Formatting and UI can be adjusted and improved upon

image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,6 +9,9 @@
     "eject": "react-scripts eject"
   },
   "dependencies": {
+    "@emotion/react": "^11.11.1",
+    "@emotion/styled": "^11.11.0",
+    "@mui/joy": "^5.0.0-beta.6",
```
Comment (wabtytsai) [inline package.json:14]: so we are using MUI now? i think we should stick to one component library so its not confusing and we are not mix-n-matching stuff from different libraries if we can help it.

File: src/apps/sports-grid/index.tsx [hunk 2]
```diff
@@ -13,11 +13,8 @@ type Props = {
 export default function Grid({ sports }: Props) {
     const gameService = useRef<Game>(getSportsGame(sports));
 
-    const puzzle = useMemo<Puzzle>(() => gameService.current.getPuzzle(), []);
-    const playerNames = useMemo<string[]>(() => {
-        const players = gameService.current.getPlayers();
-        return players.map(player => player.player_name);
-    }, []);
+    const puzzle: Puzzle = useMemo<Puzzle>(() => gameService.current.getPuzzle(), []);
+    const players: Player[] = gameService.current.getPlayers();
```
Comment (wabtytsai) [inline src/apps/sports-grid/index.tsx]: put this in useMemo to cache it, otherwise it will refetch every render

**Review by wabtytsai — APPROVED** [review]
i think the behavior for "incorrect" is, well, incorrect. I think we want to pop-up a toast or something to show it is incorrect, but still allow the users to keep guessing. Right now it's just disabled. 

accept for now, lets update that next

File: package.json (hunk 1 above)
Comment (tianshi-li) [reply package.json:14]: the other autocomplete component was too simple and could not do a couple of the things we needed

Comment (wabtytsai) [reply package.json:14]: yeah that's fine, but we should migrate off of react-bootstrap to use only MUI, unless there is some reason MUI doesn't have the components you need.

**Pull request merged**